src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -2957,99 +2956,40 @@ Real64 GLHESlinky::calcHXResistance(EnergyPlusData &state)
 
 //******************************************************************************
 
-Real64 GLHEBase::interpGFunc(Real64 const LnTTsVal // The value of LN(t/TimeSS) that a g-function
-) const
+Real64 GLHEBase::interpGFunc(Real64 const LnTTsVal) const
 {
-    // SUBROUTINE INFORMATION:
-    //       AUTHOR         Chris L. Marshall, Jeffrey D. Spitler
-    //       DATE WRITTEN   1993
-    //       MODIFIED       August, 2000
-    //       RE-ENGINEERED Dan Fisher
+    // Purpose: interpolate between g-function values, with linear extrapolation above and below range
 
-    // PURPOSE OF THIS SUBROUTINE:
-    //    To interpolate or extrapolate data in GFILE
-    //    to find the correct g-function value for a
-    //    known value of the natural log of (T/Ts)
+    auto &x = this->myRespFactors->LNTTS;
+    auto &y = this->myRespFactors->GFNC;
 
-    //  REFERENCE:          Thermal Analysis of Heat Extraction
-    //                      Boreholes.  Per Eskilson, Dept. of
-    //                      Mathematical Physics, University of
-    //                      Lund, Sweden, June 1987.
+    auto x_begin = x.begin();
+    auto x_end = x.end();
+    double x_val = LnTTsVal;
```

Comment:
Is it worth using references for any of these?  I'm not sure how heavy an iterator is, but usually we use references for anything heavier than POD.  Also the references could be const I think just to make it clear.